torch_xla/distributed/xrt_init.py

```diff
@@ -0,0 +1,241 @@
+# Copyright Amazon Web Services and its Affiliates. All Rights Reserved.
+# ==============================================================================
+
+import os
+import socket
+import subprocess
+import torch.distributed as dist
+import torch_xla.core.xla_model as xm
+import torch_xla.core.xla_env_vars as xenv
+from torch_xla.utils.utils import get_free_tcp_ports
+from torch_xla.distributed.xla_multiprocessing import _get_devices_per_worker
+
+_NODE_LIST = None
+_TCP_STORE = None
+_INIT_XRT_ALREADY_CALLED = False
+XRT_SERVER_REGEX = 'torch_neuronx.distributed._xrt_run_server'
```

Comment:
I assume most of our users don't have this installed. Should this be configurable so they can select the default server in PT/XLA instead?

https://github.com/pytorch/xla/blob/master/torch_xla/core/xrt_run_server.py

Reply:
The same server described in this module is run by rank 0 worker on a node. This is spawned by SLURM. So, as described above, in addition to the subprocess.Popen call in this module, we also spawn another "Watch" /"Polling" process. That is the only difference. (Forgot to add the module here). But pasting the main contents below:
```
polling_process = multiprocessing.Process(target=_polling,
                                              args=(int(args.pid_to_track), ))
server_process = multiprocessing.Process(
        target=torch_xla._XLAC._run_xrt_local_service, args=(int(args.port), ))
```